highlight curl and Docker

DanRoscigno opened the pull request:
- Leads with curl
- Highlights Docker
- Moves requirements to end

depends on https://github.com/ClickHouse/ClickHouse/pull/44377

closes https://github.com/ClickHouse/clickhouse-docs/issues/631
### Changelog category (leave one):
- Documentation (changelog entry is not required)

DanRoscigno committed:
highlight curl and Docker

alexey-milovidov commented:
We dropped the requirement for SSE 4.2 recently: https://s3.amazonaws.com/clickhouse-builds/0/95d115a8bba2954aae223309aac46c3b07296dd9/clickhouse_special_build_check/report.html

@rschu1ze Please edit the universal script to detect SSE 2: docs/_includes/install/universal.sh

alexey-milovidov commented:
The instruction should emphasize easy installation and portability:
ClickHouse runs anywhere; it has no dependencies and virtually no system requirements.

Let's cut down the page a little.

rschu1ze commented on docs/en/getting-started/install.md:
```diff
@@ -8,8 +8,9 @@ slug: /en/install
 You have two options for getting up and running with ClickHouse:
 
-- **[ClickHouse Cloud](https://clickhouse.com/cloud/):** the official ClickHouse as a service, - built by, maintained, and supported by the creators of ClickHouse
+- **[ClickHouse Cloud](https://clickhouse.com/cloud/):** The official ClickHouse as a service, - built by, maintained and supported by the creators of ClickHouse
```
L. 9: "Two" --> "Three"

rschu1ze commented on docs/en/getting-started/install.md:
```diff
@@ -406,3 +347,65 @@ SELECT 1
+
+### CPU Architecture
+
+ClickHouse can run on any Linux, FreeBSD, or Mac OS X with x86_64, AArch64, or PowerPC64LE CPU architecture.
```
Minor: "x86_64" --> "x86-64"

rschu1ze commented on docs/en/getting-started/install.md:
```diff
@@ -406,3 +347,65 @@ SELECT 1
+
+ClickHouse can run on any Linux, FreeBSD, or Mac OS X with x86_64, AArch64, or PowerPC64LE CPU architecture.
+
+Official pre-built binaries are typically compiled for x86_64 and leverage SSE 4.2 instruction set, so unless otherwise stated usage of CPU that supports it becomes an additional system requirement. Here’s the command to check if current CPU has support for SSE 4.2:
```
I'd consider the SSE level on x86 a detail which users should not have to worry about. Besides that, SSE4.2 was released in 2006 and there exist only very very few systems without it these days.

More importantly, with #44366, I have changed the "universal.sh" script (that should be the default recommendation) such that it now fetches the right SSE build on x86 independently of whether the system supports SSE4.2 or not.

@DanRoscigno Suggest to
- rewrite l. 357 as "Official binaries are available for x86-64 and AArch64.",
- remove l. 358-363,
- Instead add a general sentence à la "It is also possible to build ClickHouse from source, see [...] for details."

DanRoscigno replied:
Thanks @rschu1ze , based on the above, can I remove the second sentence of line #260 also:

> You can compile packages and install them or use programs without installing packages. Also by building manually you can disable SSE 4.2 requirement or build for AArch64 CPUs.

DanRoscigno committed:
apply Robert's edits

DanRoscigno committed:
add easy button

rschu1ze committed:
AArch64 --> ARM, x86_64 --> x86-64, + more

- revert back to "two" options instead of "three". Docker is a form of self-hosted ClickHouse and it is also listed as such in the text.

DanRoscigno committed:
Merge branch 'master' into simplify-install-docs

DanRoscigno committed:
Merge branch 'master' into simplify-install-docs
